### PR TITLE
Adding guest checkout

### PR DESCRIPTION
**Pull request by li-helen**
### Assignee Tasks

* [ ] added unit tests (or none needed)
* [ ] written relevant docs (or none needed)
* [ ] referenced any relevant issues (or none exist)

### Guidelines

Please add a description of this Pull Request's motivation, scope, outstanding issues or potential alternatives, reasoning behind the current solution, and any other relevant information for posterity.

---

TL;DR: 
- the routes in api/cart.js are now able to handle both user and guest carts
- the cart component itself on the client side got too complicated so i broke it out into 2 different carts that render conditionally based on whether a user is logged in

### CODE REVIEW

File: client/components/checkout.js [hunk 1]
```diff
@@ -31,7 +31,11 @@ class DisconnectedCheckout extends React.Component {
     })
 
     axios.post('/api/orders', {orderItems: orders})
-    axios.delete('/api/cart', {data: {userId: this.props.user.id}})
+    if(this.props.user) {
+        axios.delete('/api/cart', {data: {userId: this.props.user.id}})
+    } else {
+
```
Comment (tryding) [inline client/components/checkout.js]: is this supposed to be empty?

File: client/components/cart-wrapper.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
+import React from 'react'
+import {connect} from 'react-redux'
+import {Cart} from './cart'
+import {GuestCart} from './cart-guest'
+
+class DisconnectedCartWrapper extends React.Component {
+    render() {
+        console.log('INSIDE CART WRAPPER, USER IS NOW: ', this.props.user)
```
Comment (tryding) [inline client/components/cart-wrapper.js]: Stray console.log

File: client/components/cart-item.js [hunk 3]
```diff
@@ -1,6 +1,7 @@
 import React from 'react'
 
 export const CartItem = (props) => {
+    console.log('CART ITEM PROPS:', props)
```
Comment (tryding) [inline client/components/cart-item.js]: Stray console.log

File: client/components/checkout.js (hunk 1 above)
Comment (li-helen) [reply client/components/checkout.js]: ooh good catch! i was in the middle of figuring this out, and forgot to get rid of this else case after i realized i didn't need to send any data with the request if i was dealing with a guest

**Review by tryding — APPROVED** [review]

**Pull request merged**